// react-telegram-auth/lib/index.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TLoginButton = exports.TLoginButtonSize = void 0;
var React = __importStar(require("react"));
var react_1 = require("react");
var TLoginButtonSize;
(function (TLoginButtonSize) {
    TLoginButtonSize["Large"] = "large";
    TLoginButtonSize["Medium"] = "medium";
    TLoginButtonSize["Small"] = "small";
})(TLoginButtonSize = exports.TLoginButtonSize || (exports.TLoginButtonSize = {}));
var TLoginButton = /** @class */ (function (_super) {
    __extends(TLoginButton, _super);
    function TLoginButton() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this._containerRef = (0, react_1.createRef)();
        return _this;
    }
    TLoginButton.prototype.componentDidMount = function () {
        var _a = this.props, botName = _a.botName, buttonSize = _a.buttonSize, cornerRadius = _a.cornerRadius, requestAccess = _a.requestAccess, usePic = _a.usePic, onAuthCallback = _a.onAuthCallback, redirectUrl = _a.redirectUrl, lang = _a.lang;
        if (onAuthCallback != null) {
            window.TelegramOnAuthCb = function (user) { return onAuthCallback(user); };
        }
        var script = document.createElement('script');
        script.src = 'https://telegram.org/js/telegram-widget.js?21';
        script.async = true;
        script.setAttribute('data-telegram-login', botName);
        if (buttonSize != null) {
            script.setAttribute('data-size', buttonSize);
        }
        if (buttonSize != null) {
            script.setAttribute('data-radius', "".concat(cornerRadius));
        }
        if (usePic != null) {
            script.setAttribute('data-userpic', "".concat(usePic));
        }
        if (lang != null) {
            script.setAttribute('data-lang', lang);
        }
        if (redirectUrl != null) {
            script.setAttribute('data-auth-url', redirectUrl);
        }
        if (onAuthCallback != null) {
            script.setAttribute('data-onauth', 'TelegramOnAuthCb(user)');
        }
        if (requestAccess != null) {
            script.setAttribute('data-request-access', requestAccess);
        }
        // componentDidMount will only be called once after the first render.
        this._containerRef.current.appendChild(script);
    };
    TLoginButton.prototype.render = function () {
        var _a;
        return React.createElement("div", { className: "tlogin-button ".concat((_a = this.props.additionalClassNames) !== null && _a !== void 0 ? _a : ''), ref: this._containerRef });
    };
    return TLoginButton;
}(react_1.Component));
exports.TLoginButton = TLoginButton;
